### PR TITLE
GFORMS-2389 - Make all properties that accept true/false or yes/no to…

### PR DESCRIPTION
**Pull request by DursunGundogan**
… be boolean

### CODE REVIEW

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/DraftRetrievalMethod.scala [hunk 1]
```diff
@@ -17,26 +17,19 @@
 package uk.gov.hmrc.gform.sharedmodel.formtemplate
 
 import play.api.libs.json._
-import uk.gov.hmrc.gform.formtemplate.FormComponentMakerService.{ IsFalseish, IsTrueish }
 
 sealed trait DraftRetrievalMethod
 
 case class OnePerUser(continueOrDeletePage: ContinueOrDeletePage) extends DraftRetrievalMethod
 case class FormAccessCodeForAgents(continueOrDeletePage: ContinueOrDeletePage) extends DraftRetrievalMethod
-case object BySubmissionReference extends DraftRetrievalMethod
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/DraftRetrievalMethod.scala:26]: Why is this removed?

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ContinueOrDeletePage.scala [hunk 2]
```diff
@@ -28,9 +27,9 @@ object ContinueOrDeletePage {
...
   implicit val format: OFormat[ContinueOrDeletePage] = derived.oformat()
 
-  def unapply(str: String): Option[ContinueOrDeletePage] = str match {
-    case IsFalseish() => Some(Skip)
-    case IsTrueish()  => Some(Show)
-    case _            => None
+  def unapply(str: Boolean): Option[ContinueOrDeletePage] = if (str) {
+    Some(Show)
+  } else {
+    Some(Skip)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ContinueOrDeletePage.scala:33]: This never returns `None` so there is no need for `Option[ContinueOrDeletePage]` as a return type

**Review by josef-vlach — CHANGES_REQUESTED** [review]

**Comment by josef-vlach** [comment]
Also would it be possible to get rid of these objects?

```
object IsTrueish
object IsFalseish
final object IsDisplayCharCountFalse
final object IsDisplayCharCountTrue
```

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ContinueOrDeletePage.scala (hunk 2 above)
Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ContinueOrDeletePage.scala:33]: There is a pattern matching [here](https://github.com/hmrc/gform/blob/main/app/uk/gov/hmrc/gform/sharedmodel/formtemplate/DraftRetrievalMethod.scala#L31) I think we should return it as Option

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ContinueOrDeletePage.scala:33]: > There is a pattern matching [here](https://github.com/hmrc/gform/blob/main/app/uk/gov/hmrc/gform/sharedmodel/formtemplate/DraftRetrievalMethod.scala#L31) I think we should return it as Option

In provided link we can get rid of matching on `showContinueOrDeletePage: Boolean` and use id directly.

So we should provide this 

```
def fromBoolean(str: Boolean): ContinueOrDeletePage = if (str) Show else Skip
```
instead of this `unapply` method

**Review by josef-vlach — APPROVED** [review]
